bench/wtperf/wtperf.c

```diff
@@ -1270,26 +1268,14 @@ backup_worker(void *arg)
 
         wtperf->backup = true;
 
-        /*
-         * open_cursor can return EBUSY if concurrent with a metadata operation, retry in that case.
-         */
-        while (
-          (ret = session->open_cursor(session, "backup:", NULL, NULL, &backup_cursor)) == EBUSY)
-            __wt_yield();
-        if (ret != 0)
-            goto err;
-
-        while ((ret = backup_cursor->next(backup_cursor)) == 0) {
-            testutil_check(backup_cursor->get_key(backup_cursor, &key));
-            backup_read(wtperf, key);
+        /* Take the kind of backup configured. */
+        if (opts->backup_complete == 0)
+            backup_read(wtperf, session);
+        else {
+            testutil_backup_create_full(
+              conn, wtperf->home, (int)thread->backup.ops, false, 1024, NULL);
+            testutil_delete_old_backups(4);
```

Comment:
Why we need to retain 4 backups? One is not enough?

Reply:
In the testing we want to keep some number of old backups so that if there is a problem we can look back and see what earlier backups contain. For my project I need to compare the differences between backups to know how much changed between them for analysis. So one is enough to restore from but not enough for analysis.